### PR TITLE
rbd: add snap_exists method API

### PR DESCRIPTION
**Pull request by zhengchengyao**
Signed-off-by: Zheng Yin <zhengyin@cmss.chinamobile.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/include/rbd/librbd.h [hunk 1]
```diff
@@ -763,6 +763,7 @@ CEPH_RBD_API int rbd_deep_copy_with_progress(rbd_image_t image,
 CEPH_RBD_API int rbd_snap_list(rbd_image_t image, rbd_snap_info_t *snaps,
                                int *max_snaps);
 CEPH_RBD_API void rbd_snap_list_end(rbd_snap_info_t *snaps);
+CEPH_RBD_API int rbd_snap_exists2(rbd_image_t image, const char *snapname, bool *exists);
```
Comment (dillaman) [inline src/include/rbd/librbd.h]:
```suggestion
CEPH_RBD_API int rbd_snap_exists(rbd_image_t image, const char *snapname, bool *exists);
```

File: src/librbd/librbd.cc [hunk 2]
```diff
@@ -5292,6 +5292,16 @@ extern "C" int rbd_snap_get_limit(rbd_image_t image, uint64_t *limit)
   return r;
 }
 
+extern "C" int rbd_snap_exists2(rbd_image_t image, const char *snapname, bool *exists)
```
Comment (dillaman) [inline src/librbd/librbd.cc]:
```suggestion
extern "C" int rbd_snap_exists(rbd_image_t image, const char *snapname, bool *exists)
```

File: src/pybind/rbd/rbd.pyx [hunk 3]
```diff
@@ -502,6 +502,7 @@ cdef extern from "rbd/librbd.h" nogil:
     int rbd_snap_unprotect(rbd_image_t image, const char *snap_name)
     int rbd_snap_is_protected(rbd_image_t image, const char *snap_name,
                               int *is_protected)
+    int rbd_snap_exists2(rbd_image_t image, const char *snapname, bint *exists)
```
Comment (dillaman) [inline src/pybind/rbd/rbd.pyx]:
```suggestion
    int rbd_snap_exists(rbd_image_t image, const char *snapname, bint *exists)
```

**Comment by dillaman** [comment]
Nit: you have a typo (`meathod`) in your second commit subject line

**Comment by zhengchengyao** [comment]
> Nit: you have a typo (`meathod`) in your second commit subject line

thanks, Jason, it was careless.

**Comment by zhengchengyao** [comment]
jenkins test docs

**Comment by dillaman** [comment]
jenkins test make check

**Comment by zhengchengyao** [comment]
hi, @dillaman, I notice that make chek (arm64) failed, some Packages are not found:

Error: No Package found for pyOpenSSL3
Error: No Package found for python3-bcrypt
Error: No Package found for python3-cherrypy
Error: No Package found for python3-coverage
Error: No Package found for python3-jwt
Error: No Package found for python3-nose
Error: No Package found for python3-pecan
Error: No Package found for python3-pyOpenSSL
Error: No Package found for python3-requests
Error: No Package found for python3-routes
Error: No Package found for python3-six
Error: No Package found for python3-tox
Error: No Package found for python3-virtualenv
Error: No Package found for python3-werkzeug
Build step 'Execute shell' marked build as failure

**Comment by dillaman** [comment]
The arm64 builds have been broken for a long time now so it's just usually ignored.

**Pull request merged**